stream-chat-android-client/src/main/java/io/getstream/chat/android/client/api2/model/dto/UpstreamMessageDto.kt

```diff
@@ -1,17 +1,35 @@
 package io.getstream.chat.android.client.api2.model.dto
 
 import com.squareup.moshi.JsonClass
+import java.util.Date
 
 @JsonClass(generateAdapter = true)
 internal data class UpstreamMessageDto(
     val id: String,
     val cid: String,
+    val parent_id: String?,
```

Comment:
I think that adding @Json(name = "[field name]") to all fields woudl be nice, so we prevent proguard problems. Just making sure that the `[...]/api2/model/dto` is in the proguard rules works too.

Reply:
I prefer doing both =P

Reply:
We don't have these annotations on purpose, see https://github.com/GetStream/stream-chat-android/pull/1274#discussion_r564593801

As for Proguard: Moshi's generated adapters contain the string names of the fields as strings, so obfuscation shouldn't affect them at all.

Here's an example of the trie that Moshi uses for figuring out which field to parse being created in the generated code: 
![image](https://user-images.githubusercontent.com/12054216/106474490-42ead600-64a5-11eb-81d5-7e8a632335cd.png)

Reply:
Makes sense. Solving the conversation.